vyper/semantics/types/function.py

```diff
@@ -327,10 +327,15 @@ def from_vyi(cls, funcdef: vy_ast.FunctionDef) -> "ContractFunctionT":
         -------
         ContractFunctionT
         """
-        function_visibility, state_mutability, nonreentrant = _parse_decorators(funcdef)
+        function_visibility, state_mutability, nonreentrant = _parse_decorators(
+            funcdef, is_interface=True
+        )
 
-        if nonreentrant:
-            raise FunctionDeclarationException("`@nonreentrant` not allowed in interfaces", funcdef)
+        assert not nonreentrant
+
+        # it's redundant to specify visibility in vyi - always should be external
+        function_visibility = function_visibility or FunctionVisibility.EXTERNAL
```

Comment:
more explicit:
```suggestion
        if function_visibility is None:
            FunctionVisibility.EXTERNAL
```

Reply:
addressed in: a6f1a86ae7f3a6c920682e1b30c8675244b4a504